### PR TITLE
Fix PHP 8 incompatibilities

### LINKED ISSUE #1361: Fix PHP 8 incompatibilities
## Describe the bug
It has come to our attention that we have some code that is incompatible with PHP 8.x. This makes the Content Helper Sidebar's "Performance Details" panel return an `Error: The response is not a valid JSON response.` message.

## To Reproduce
Try using the CHS Performance Details panel with PHP 8 or above.

## Expected behavior
The Content Helper should work as expected (fetch results and show them).

### PR DESCRIPTION
**Pull request by acicovic**
## Description
We have found parts of our code which are incompatible with PHP 8.x.

## Motivation and context
- Closes #1361.
- More context is available in that issue.

## How has this been tested?
1. We first introduced integration tests that would expose the incompatibilities and let GitHub run them.
2. As expected, the tests failed.
3. We committed the fixes that solve the issue.
4. The tests then passed.

### CODE REVIEW

File: tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php [hunk 1]
```diff
@@ -54,7 +52,7 @@ public function get_endpoint(): Base_API_Proxy {
 	 * @uses \Parsely\RemoteAPI\Remote_API_Base::__construct
 	 */
 	public function test_register_routes_by_default(): void {
-		parent::test_register_routes_by_default();
+		parent::run_test_register_routes_by_default();
```
Comment (acicovic) [inline tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php:55]: Renamed all related methods as such in order to reduce invalid complaints from PHPCS and messages that the methods needed a `covers` comment in the base class.

File: tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php [hunk 2]
```diff
@@ -123,7 +146,24 @@ public function test_get_items(): void {
...
...hwartz and Andrew Butler"], "full_content_word_count": 1785, "image_url": "https://blog.parse.ly/wp-content/uploads/2021/05/pexels-brett-jordan-998501-1024x768-2.jpeg?w=150&h=150&crop=1", "link": "https://blog.parse.ly/5-tagging-best-practices-content-strategy/?itm_source=parsely-api", "metadata": "", "metrics": {"views": 40}, "pub_date": "2021-04-30T20:30:24", "section": "Analytics That Matter", "tags": ["parsely_smart:entity:Analytics", "parsely_smart:entity:Best practice", "parsely_smart:entity:Hashtag", "parsely_smart:entity:Metadata", "parsely_smart:entity:Search engine", "parsely_smart:entity:Search engine optimization", "parsely_smart:entity:Tag (metadata)", "parsely_smart:iab:Business", "parsely_smart:iab:Science", "parsely_smart:iab:Software", "parsely_smart:iab:Technology"], "thumb_url_medium": "https://images.parsely.com/ap3YSufqxnLpz6zzQshoks3snXI=/85x85/smart/https%3A//blog.parse.ly/wp-content/uploads/2021/05/pexels-brett-jordan-998501-1024x768-2.jpeg%3Fw%3D150%26h%3D150%26crop%3D1", "title": "5 Tagging Best Practices For Getting the Most Out of Your Content Strategy", "url": "https://blog.parse.ly/5-tagging-best-practices-content-strategy/?itm_source=parsely-api"}]}',
```
Comment (acicovic) [inline tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php:126]: Reformatted this to make it readable. This has been done in other parts of the code as well.

File: tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php [hunk 3]
```diff
@@ -139,7 +179,7 @@ function () use ( &$dispatched ) {
 						'author'         => 'Aakash Shah',
 						'date'           => wp_date( $date_format, strtotime( '2020-04-06T13:30:58' ) ),
 						'id'             => 'https://blog.parse.ly/web-analytics-software-tools/?itm_source=parsely-api',
-						'dashUrl'        => PARSELY::DASHBOARD_BASE_URL . '/blog.parsely.com/find?url=https%3A%2F%2Fblog.parse.ly%2Fweb-analytics-software-tools%2F%3Fitm_source%3Dparsely-api',
```
Comment (acicovic) [inline tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php:142]: This was a bug in the test, coming from the fact that we were separately setting the Site ID and API Secret above.

File: tests/Integration/ProxyEndpointTest.php [hunk 4]
```diff
@@ -156,4 +155,35 @@ function () use ( &$dispatched ) {
 			$error->get_error_message()
 		);
 	}
+
+	/**
+	 * Verifies that calls return an error and do not perform a remote call when
+	 * the API Secret is not populated in site options.
+	 */
+	public function run_test_get_items_fails_without_api_secret_set() {
```
Comment (acicovic) [inline tests/Integration/ProxyEndpointTest.php:163]: Introduced a method to test API Secret

File: tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php (hunk 2 above)
Comment (mehmoodak) [reply tests/Integration/Endpoints/AnalyticsPostsProxyEndpointTest.php:126]: 👍

**Review by mehmoodak — APPROVED** [review]

**Comment by mehmoodak** [comment]
New commit looks good 👍.

**Pull request merged**